src/adc.rs

```diff
@@ -14,27 +17,95 @@ use crate::{
         adc::{Channel as EmbeddedHalChannel, OneShot},
         blocking::delay::DelayUs,
     },
-    pac,
+    pac::{self, ADC1},
     rcc::{Enable, Reset, AHB2, CCIPR},
     signature::{VrefCal, VtempCalHigh, VtempCalLow, VDDA_CALIB_MV},
 };
 
-use pac::{ADC1, ADC_COMMON};
-use stable_deref_trait::StableDeref;
+/// Internal voltage reference channel, used for calibration.
+pub struct Vref {
+    _0: (),
+}
+
+/// Internal battery monitoring channel.
+pub struct Vbat {
+    _0: (),
+}
+
+/// Internal temperature sensor channel.
+pub struct Temperature {
+    _0: (),
+}
+
+/// Wrapper for safely sharing [`ADC_COMMON`](pac::ADC_COMMON) between `Adc`s.
+#[derive(Clone, Copy)]
+pub struct AdcCommon {
+    _0: PhantomData<pac::ADC_COMMON>,
```

Comment:
What does this `_0` do here? The `ADC_COMMON` is not a generic so the `PhantomData` is not needed.

Reply:
I added these so they behave as if they contain these types. This may or may not be needed though.